src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt

```diff
@@ -402,11 +398,26 @@ internal class RecommendationService(
     refreshData(pageRefreshIds, existingRecommendationEntity.data)
 
     val result = updateAndSaveRecommendation(existingRecommendationEntity, userId, readableUserName)
```

Comment:
You can avoid creating the `result` variable here by using the `also` [scope function](https://kotlinlang.org/docs/scope-functions.html)

```kotlin
return updateAndSaveRecommendation(existingRecommendationEntity, userId, readableUserName).also {
  log.info("recommendation for ${it.crn} updated for recommendationId $recommendationId")

  if (sendManagementOversightDomainEvent) {
    log.info("send spo rationale to delius")
    sendManagementOversightDomainEvent(recommendationId, existingRecommendationEntity, userId)
  }

  if (sendConsiderationRationaleToDelius) {
    log.info("send consideration rationale to delius")
    sendConsiderationRationaleDomainEvent(recommendationId, existingRecommendationEntity, userId)
  }

  if (sendSpoDeleteRationaleToDelius) {
    log.info("send spo delete recommendation rationale to delius")
    sendDeleteRecommendationDomainEvent(recommendationId, existingRecommendationEntity, userId)
  }

  val sendRecommendationStartedDomainEvent =
    featureFlags?.flagDomainEventRecommendationStarted == true && featureFlags.flagConsiderRecall == true &&
        existingRecommendationEntity.data.recommendationStartedDomainEventSent != true

  if (sendRecommendationStartedDomainEvent) {
    log.info("About to send domain event for ${existingRecommendationEntity.data.crn} on Recommendation started for recommendation id $recommendationId")
    sendRecommendationStartedEvent(existingRecommendationEntity.data.crn)
    log.info("Sent domain event for ${existingRecommendationEntity.data.crn} on Recommendation started asynchronously for recommendation id $recommendationId")
    existingRecommendationEntity.data.recommendationStartedDomainEventSent = true
  }
}
```

Reply:
Hmm, this is interesting but I don't really like the look of it, and its readability.